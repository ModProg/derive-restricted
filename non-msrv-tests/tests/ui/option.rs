use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Default)]
struct StructNoOption<T>(#[derive_where] PhantomData<T>);

#[derive_where(Default)]
struct StructWrongSyntax<T>(#[derive_where = "default"] PhantomData<T>);

#[derive_where(Default)]
enum EnumNoOption<T> {
	#[derive_where]
	A(PhantomData<T>),
}

#[derive_where(Default)]
enum EnumWrongSyntax<T> {
	#[derive_where = "default"]
	A(PhantomData<T>),
}

#[derive_where(Clone)]
struct StructInvalidOption<T>(#[derive_where(option)] PhantomData<T>);

#[derive_where(Clone)]
enum EnumInvalidOption<T> {
	#[derive_where(option)]
	A(PhantomData<T>),
}

fn main() {}
