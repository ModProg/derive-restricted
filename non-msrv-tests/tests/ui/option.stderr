error: unexpected option syntax
 --> tests/ui/option.rs:4:28
  |
4 | struct StructNoOption<T>(#[derive_where] PhantomData<T>);
  |                            ^^^^^^^^^^^^

error: unexpected option syntax
 --> tests/ui/option.rs:7:31
  |
7 | struct StructWrongSyntax<T>(#[derive_where = "default"] PhantomData<T>);
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/option.rs:11:4
   |
11 |     #[derive_where]
   |       ^^^^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/option.rs:17:4
   |
17 |     #[derive_where = "default"]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^

error: unknown option
  --> tests/ui/option.rs:22:46
   |
22 | struct StructInvalidOption<T>(#[derive_where(option)] PhantomData<T>);
   |                                              ^^^^^^

error: unknown option
  --> tests/ui/option.rs:26:17
   |
26 |     #[derive_where(option)]
   |                    ^^^^^^
